src/main/java/ru/javawebinar/topjava/web/MealServlet.java

```diff
@@ -0,0 +1,74 @@
+package ru.javawebinar.topjava.web;
+
+import ru.javawebinar.topjava.model.Meal;
+import ru.javawebinar.topjava.service.MealService;
+import ru.javawebinar.topjava.service.MealServiceImplMem;
+import ru.javawebinar.topjava.util.MealsUtil;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import org.slf4j.Logger;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class MealServlet extends HttpServlet {
+
+    private static final Logger LOG = getLogger(MealServlet.class);
+
+    private MealService mealService;
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        mealService = new MealServiceImplMem();
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        request.setCharacterEncoding("UTF-8");
+        String sId = request.getParameter("id");
+        Meal meal = new Meal(sId.isEmpty() ? null : Integer.valueOf(sId),
+                LocalDateTime.parse(request.getParameter("dateTime")),
+                request.getParameter("description"),
+                Integer.valueOf(request.getParameter("calories")));
+        mealService.save(meal);
+        LOG.debug("meal " + meal.toString() + " saved");
+        response.sendRedirect("meals");
+    }
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String action = request.getParameter("action");
+        if (action == null)
+            action = "all";
```

Comment:
I meant you could use unary operator, which was inside switch brackets. I don't remember logic, but like 
String action = request.getparameter("action") == null ? "all" : something; If instead of something should be null, then okey.
And second point is if(action == null) action = "all"; without curly brackets. It also correct, but I'm fan to use  curly brackets always after if or else or for...

Reply:
So it's for the future